internal/storage/bucket_handle.go

```diff
@@ -465,6 +464,54 @@ func (b *bucketHandle) ComposeObjects(ctx context.Context, req *gcs.ComposeObjec
 	return
 }
 
+func (b *bucketHandle) DeleteFolder(ctx context.Context, folderName string) (err error) {
+	if b.BucketType() == gcs.NonHierarchical {
+		err = b.DeleteObject(
+			ctx,
+			&gcs.DeleteObjectRequest{
+				Name:       folderName,
+				Generation: 0, // Delete the latest version of object named after dir.
+			})
+
+		if err != nil {
+			err = fmt.Errorf("DeleteObject: %w", err)
+		}
+	} else if b.BucketType() == gcs.Hierarchical {
+		err = b.deleteFolderForHierarchicalBucket(ctx, folderName)
+	}
+
+	return err
+}
+
+func (b *bucketHandle) deleteFolderForHierarchicalBucket(ctx context.Context, folderName string) (err error) {
```

Comment:
Ideally an error should be return indicating unable to delete folder as there is an item inside it in case of HNS, please test this scenario in manual testing for flat and hns both as part of delete functionality.

Reply:
Yes getting `unable to delete folder ` because of `Directory not empty`.